CONTRIBUTING.md

```diff
@@ -2,7 +2,52 @@
 
 This document provides a collection of rules and guidelines that need to be adhered to when working with or contributing to FluidMS.
 
-## Commit guidelines
+## Git workflow
+
+Before you submit a pull request, please create an issue first so we can discuss the need for this specific feature and clarify the details of the implementation before you invest your time in any work that is potentially wasted.
```

Comment:
Not sure about the contribution workflow in other OSS projects but I'd recommend this only for larger PRs. 
Smaller ones can be discussed in the PR otherwise it might "scare off" contributors if they need to file an issue first, discuss it and then potentially work on the code.
This might raise the bar for contributions.

But then again, what is considered "small" and "large"

Reply:
> otherwise it might "scare off" contributors

Good point! I’ll take out the whole paragraph.